src/psyclone/psyGen.py

```diff
@@ -5806,56 +5848,200 @@ def _number_handler(self, node, parent):
+    to that symbol (i.e. the mechanism by which it is accessed).
 
     :param str name: Name of the symbol.
-    :param str datatype: Data type of the symbol.
+    :param str datatype: Data type of the symbol. (One of \
+                     :py:attr:`psyclone.psyGen.Symbol.valid_data_types`.)
     :param list shape: Shape of the symbol in column-major order (leftmost \
                        index is contiguous in memory). Each entry represents \
                        an array dimension. If it is 'None' the extent of that \
                        dimension is unknown, otherwise it holds an integer \
                        literal or a reference to an integer symbol with the \
                        extent. If it is an empty list then the symbol \
                        represents a scalar.
-    :param str scope: It is 'local' if the symbol just exists inside the \
-                      kernel scope or 'global_*' if the data survives outside \
-                      of the kernel scope. Note that global-scoped symbols \
-                      also have postfixed information about the sharing \
-                      mechanism, at the moment just 'global_argument' is \
-                      available for variables passed in/out of the kernel \
-                      by argument.
+    :param interface: Object describing the interface to this symbol (i.e. \
+                      whether it is passed as a routine argument or accessed \
+                      in some other way) or None if the symbol is local.
+    :type interface: :py:class:`psyclone.psyGen.SymbolInterface` or NoneType.
     :param constant_value: Sets a fixed known value for this \
                            Symbol. If the value is None (the default) \
                            then this symbol is not a constant. The \
                            datatype of the constant value must be \
                            compatible with the datatype of the symbol.
     :type constant_value: int, str or bool
-    :param bool is_input: Whether the symbol represents data that exists \
-                          before the kernel is entered and that is passed \
-                          into the kernel.
-    :param bool is_output: Whether the symbol represents data that is passed \
-                           outside the kernel upon exit.
 
     :raises NotImplementedError: Provided parameters are not supported yet.
     :raises TypeError: Provided parameters have invalid error type.
     :raises ValueError: Provided parameters contain invalid values.
 
     '''
-
-    # Tuple with the valid values for the access attribute.
-    valid_scope_types = ('local', 'global_argument')
-    # Tuple with the valid datatypes.
-    valid_data_types = ('real', 'integer', 'character', 'boolean')
-    # Mapping from supported data types for constant values to
-    # internal Python types
+    ## Tuple with the valid datatypes.
```

Comment:
Hmm. Not sure what to do about that. The reason I've added the '##' is because that's Doxygen mark-up and it means our comments get picked up in the Doxygen-generated reference guide.
I think we should discuss this in the telco tomorrow. Perhaps we can disable that particular pycodestyle error?